src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs

```diff
@@ -830,5 +830,46 @@ public void TransposeInto(ref Block8x8F d)
                 d.V7R.W = this.V7R.W;
             }
         }
+
+        /// <summary>
+        /// Compares entire 8x8 block to a single scalar value.
+        /// </summary>
+        /// <param name="value">Value to compare to.</param>
+        public bool EqualsToScalar(int value)
+        {
+#if SUPPORTS_RUNTIME_INTRINSICS
+            if (Avx2.IsSupported)
+            {
+                const int equalityMask = unchecked((int)0b1111_1111_1111_1111_1111_1111_1111_1111);
+
+                var targetVector = Vector256.Create(value);
+                ref Vector256<float> blockStride = ref this.V0;
+
+                for (int i = 0; i < RowCount; i++)
+                {
+                    Vector256<int> areEqual = Avx2.CompareEqual(Avx.ConvertToVector256Int32WithTruncation(Unsafe.Add(ref this.V0, i)), targetVector);
```

Comment:
```suggestion
                    Vector256<int> areEqual = Avx2.CompareEqual(Avx.ConvertToVector256Int32WithTruncation(Unsafe.Add(ref this.V0, (nint)(uint)i)), targetVector);
```
Doesn't look that nice, but avoids the `movsxd` instruction (cf. [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdwAJ4YYDALLkAFLABmDa7YYATbBmxoDLw8Qd4IAJQMALwAfAwAqly42K6CAILe3i4w7r7+gbxh4RrU5hYeNnb2pNnunnZ5AUEhhRHRcYnJqSwZWW4+fk1OXMEY4U4co+GtxTQAvkA)).

Alternatively type the loop variable as `nint` and cast `(nint)(uint)RowCount`.

Same on other places where Unsafe.Add with an int-offset is used.

Reply:
I can’t remember what language version we’re targeting. We may have to upgrade in the shared infrastructure props

Reply:
Not 9.0 for afaik, I remember trying to use `nint` and failed to do so.

Reply:
I can upgrade but I'm not convinced we should bother with the optimization here or elsewhere. It's damn ugly and I'm sure also fixed in new target frameworks.

Reply:
We can use `nint` directly for indices so we won't need to cast `(nint)(uint)`.

Reply:
LangVersion updated upstream!

`git submodule foreach git pull origin master`